tkg/client/upgrade_region.go

```diff
@@ -317,6 +318,9 @@ func (c *TkgClient) configureVariablesForProvidersInstallation(regionalClusterCl
 		// no variable configuration is needed to deploy Docker provider as
 		// infrastructure-components.yaml for docker does not require any variable
 	}
+
+	c.SetProxyEnvironmentVariable()
```

Comment:
Thanks @lubronzhan , for the question 3 above what I meant is I think this line will not be called during management cluster creation, because `configureVariablesForProvidersInstallation` will be returned above in here
https://github.com/vmware-tanzu/tanzu-framework/blob/b6c690beafcb3269ce348f7336b9f8a2ded80bde/tkg/client/upgrade_region.go#L276-L278

Since `configureVariablesForProvidersInstallation` is shared by management cluster creation and upgrade, during creation the client is `nil`

https://github.com/vmware-tanzu/tanzu-framework/blob/b6c690beafcb3269ce348f7336b9f8a2ded80bde/tkg/client/init.go#L211